### ISSUE TITLE
imphook call return inconsistencies

### ISSUE THREAD

**Issue by mr-tz** [vivisect/vivisect#515]
Is it expected that some of the default hooks perform a call return, e.g.

https://github.com/vivisect/vivisect/blob/e0592b15dcaeef4fe6c6c2f2df1f56b74ceb088e/vivisect/impemu/platarch/windows.py#L84-L88

vs. 

https://github.com/vivisect/vivisect/blob/e0592b15dcaeef4fe6c6c2f2df1f56b74ceb088e/vivisect/impemu/platarch/windows.py#L135-L138

I think this leads to some unexpected behavior during emulation.

**Comment by atlas0fd00m** [comment]
how so?  do you have any example?

**Comment by mr-tz** [comment]
As an example see the test cases in viv-utils, e.g. https://github.com/williballenthin/viv-utils/blob/4b2a529374d7a65cb542ce798e8ae313a4838307/tests/test_driver.py#L373

We had to add `vudrv.remove_default_viv_hooks(emu)` to remove the emulated APIs that do not return.

The test file is available in https://github.com/williballenthin/viv-utils/tree/master/tests/data

```asm
.text:10001010 mov     eax, 11F8h
.text:10001015 call    __alloca_probe
.text:1000101A mov     eax, [esp+11F8h+fdwReason]
```

When encountering the `__alloca_probe` the emulator does not appear to behave correctly as we end up trying to execute `0xfefefefe`.

If this is hard to follow for you I can provide an example with pure vivisect code and trace.

**Comment by mr-tz** [comment]
I still see this as inconsistent:
* some hooks (kernel32.*) call `callconv.execCallReturn` so change EIP to after the function
* other hooks (ntdll.*) change the emulator state, but EIP does not change

Is this expected and I'm understanding/using it incorrectly?
